src/ngraph/runtime/cpu/cpu_backend.cpp

```diff
@@ -82,22 +82,20 @@ bool runtime::cpu::CPU_Backend::compile(shared_ptr<Function> func)
         auto cf = instance.m_external_function->make_call_frame();
         instance.m_call_frame = dynamic_pointer_cast<CPU_CallFrame>(cf);
     }
-    return true;
+    return func;
 }
 
 std::shared_ptr<ngraph::runtime::cpu::CPU_CallFrame>
     runtime::cpu::CPU_Backend::get_call_frame(std::shared_ptr<Function> func)
 {
-    bool rc = true;
     FunctionInstance& instance = m_function_map[func];
     if (instance.m_external_function == nullptr)
     {
-        rc = compile(func);
-    }
-
-    if (!rc)
-    {
-        throw ngraph_error("couldn't compile a function");
+        auto rc = compile(func);
+        if (!rc)
+        {
+            throw ngraph_error("couldn't compile a function");
```

Comment:
Need better error messages than this. This is **extremely** important for end users.

Reply:
We have gone around and around on this. If there is an error here there is pretty much nothing that a user can do except report the error to us developers. I suppose I could change it to something like "Internal compile error, please file a bug report"
When we enabled dumping out of the compile error that was causing problems with the unit tests long ago.

Reply:
The way I am looking at is the following:
1. If the compilation error is related to nGraph language (i.e., mismatch of op/parameters etc.) they the user needs to know that so tat they can fix the framework level code to resolve.
2. If the error is internal then of course also need to know so that they can file a bug report. 

In this case the cpu backend `compile(...)` needs to propagate that error message.